### ISSUE TITLE
Checkbox: aria-posinset in not allowed attribute on input element

### ISSUE THREAD

**Issue by Anschur** [microsoft/fluentui#13483]
### Describe the issue:

The [checkbox ](https://developer.microsoft.com/en-us/fluentui#/controls/web/checkbox)control provides props for setting aria-posinset and aria-setsize. These aria attributes will be set on the input-element in the html generated. But [aria-posinset is only allowed for specific roles](https://www.w3.org/TR/2017/REC-wai-aria-1.1-20171214/#aria-posinset), and _input_ is not among them.

I do not know if there are any direct user impact of this bug, or if there are reasons for keeping this as is. It might be that Narrator reads the order, but I don't think other screen readers would as it is not according to spec. So keeping this as is might lead developers astray. 

### Please provide a reproduction of the issue in a codepen:

1. Go to https://codepen.io/svartn/pen/PoZwdGP
2. Scan the page with Accessibility Insights
3. Observe that bug is reported on the second checkbox that has aria-posinset attribute present. 

### Documentation describing expected behavior

Allowed roles for aria-posinset: https://www.w3.org/TR/2017/REC-wai-aria-1.1-20171214/#aria-posinset

**Comment by paulgildea** [comment]
@jurokapsiar - Can help with a point of view on this issue. @jdhuntington mentioned #4006 where these props were added. 

Looks they were mostly added for when they in the context of a list scenario. Is this right pattern?

Either way, if we were to remove those properties it would be breaking change - we should keep this in mind for when we converge input types.

**Comment by jurokapsiar** [comment]
There is a multi-select listbox scenario that is valid: https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-rearrangeable.html, but that uses listbox / option roles. I would need to do some tests to see if screen readers accept `<input>` based checkboxes with overriden `option` role.

It would probably be more understandable for the users to have a selectable list component with multi-select capability.

**Comment by xugao** [comment]
@Anschur - may i ask why you are setting `ariaPositionInSet="3"` and `airaSetSize="10"` in your codepen ?
These are optional props and unless you are setting a non-default role on the `input` element, you shouldn't be setting these aria attributes which caused the violations?
